### PR TITLE
Fail on deprecation warnings in tooling api cross version tests

### PR DESCRIPTION
**Pull request by reinsch82**
address https://github.com/gradle/gradle-private/issues/3814

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test aCT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle test ACT

**Comment by reinsch82** [comment]
@bot-gradle clean

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@eskatos Thx for looking into the first try on this.
I created a new PR to try something else, that seems to work better now.

File: subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy [hunk 1]
```diff
@@ -127,19 +135,47 @@ class ToolingApi implements TestRule {
...
 
-    def <T> T withConnection(GradleConnector connector, Closure<T> cl) {
+    def <T> T withConnection(connector, @DelegatesTo(ProjectConnection) Closure<T> cl) {
         return withConnectionRaw(connector, cl)
     }
 
+    def <T> T loadToolingLeanModel(Class<T> modelClass, @DelegatesTo(ModelBuilder<T>) Closure configurator = {}) {
+        def result = withConnection {
+            def builder = it.model(modelClass)
+            builder.tap(configurator)
```
Comment (donat) [inline subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy:154]: 🧐 
```suggestion
            configurator(builder)
```

Comment (reinsch82) [reply subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy:154]: The `tap` changes the `this` whereas with the call you need to use `it` to refer to the builder.
Or is there something else I'm missing?

Comment (donat) [reply subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApi.groovy:154]: You are right, thanks for the clarification.

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Comment by reinsch82** [comment]
@bot-gradle test RFN

File: subprojects/kotlin-dsl-tooling-builders/src/testFixtures/groovy/org/gradle/kotlin/dsl/tooling/builders/KotlinScriptModulParameters.groovy [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+package org.gradle.kotlin.dsl.tooling.builders
+
+
+import org.gradle.tooling.model.kotlin.dsl.KotlinDslModelsParameters
+import org.gradle.tooling.model.kotlin.dsl.KotlinDslScriptsModel
+
+import static org.gradle.integtests.fixtures.RepoScriptBlockUtil.gradlePluginRepositoryMirrorUrl
+import static org.gradle.kotlin.dsl.resolver.KotlinBuildScriptModelRequestKt.newCorrelationId
+
+class KotlinScriptModulParameters {
```
Comment (eskatos) [inline subprojects/kotlin-dsl-tooling-builders/src/testFixtures/groovy/org/gradle/kotlin/dsl/tooling/builders/KotlinScriptModulParameters.groovy:26]: :mag_right: Typo
```suggestion
class KotlinScriptModelParameters {
```

**Review by eskatos — APPROVED** [review]
LGTM

So, it seems that the Kotlin DSL script model tests don't emit deprecation warnings while we expected them to :thinking:

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**